cdap-ui/app/features/etlapps/controllers/create-ctrl.js

```diff
@@ -148,52 +148,61 @@ angular.module(PKG.name + '.feature.etlapps')
       if ($scope.source.placeHolderSource) {
         return;
       }
-      $bootstrapModal.open({
-        templateUrl: '/assets/features/etlapps/templates/create/sourceProperties.html',
-        scope: $scope,
-        backdrop: true,
-        keyboard: true
-      });
+      var filterFilter = $filter('filter'),
+          match;
+      match = filterFilter($scope.tabs, {type: 'source'});
+      if (match.length) {
+        $scope.tabs[$scope.tabs.indexOf(match[0])].active = true;
+      } else {
+        $scope.tabs.push({
+          title: $scope.source.name,
+          type: 'source',
+          active: true,
+          partial: '/assets/features/etlapps/templates/create/tabs/sourcePropertyEdit.html'
+        })
+      }
     };
     $scope.editSinkProperties = function() {
       if ($scope.sink.placeHolderSink) {
         return;
       }
-      $bootstrapModal.open({
-        templateUrl: '/assets/features/etlapps/templates/create/sinkProperties.html',
-        scope: $scope,
-        backdrop: true,
-        keyboard: true
-
-      });
 
+      var filterFilter = $filter('filter'),
+          match;
+      match = filterFilter($scope.tabs, {type: 'sink'});
+      if (match.length) {
+        $scope.tabs[$scope.tabs.indexOf(match[0])].active = true;
+      } else {
+        $scope.tabs.active = ($scope.tabs.push({
+          title: $scope.sink.name,
+          type: 'sink',
+          active: true,
+          partial: '/assets/features/etlapps/templates/create/tabs/sinkPropertyEdit.html'
+        })) -1;
```

Comment:
Array.prototype.push returns the length of the array after the push. Since push is adding it to the last we just -1 to get the index of the newly added transform.